src/sql/workbench/contrib/notebook/browser/cellViews/linkHandler.directive.ts

```diff
@@ -73,11 +74,14 @@ export class LinkHandlerDirective {
 			}
 		}
 		if (uri && this.openerService) {
+			if (uri.fragment !== '') {
```

Comment:
This check seems unnecessary - just store it here (no reason to declare fragment above either)

Also it's always a good idea to leave comments when doing stuff like this - it definitely isn't clear from the code why we'd need to do this

```
// asFileUri removes the fragment so store that now before converting
const fragment = uri.fragment;`
```